File: app/src/main/java/com/lucagiorgetti/surprix/model/Set.java
package com.lucagiorgetti.surprix.model;

import java.util.Comparator;

/**
 * Created by Utente on 17/04/2017.
 */

public class Set {
    private String id = null;
    private String code = null;
    private String name = null;
    private String year_id = null;
    private String year_desc = null;
    private int year_year = -1;
    private String producer_id = null;
    private String producer_name = null;
    private String producer_color = null;
    private String nation = null;
    private String img_path = null;
    private String category = null;
    private boolean effective_code = true;


    public Set(String name, String code, Year year, Producer producer, String nation, String img_path, String category) {
        this.name = name;
        this.code = code;
        this.producer_id = producer.getId();
        this.producer_name = producer.getName();
        this.producer_color = producer.getColor();
        this.year_id = year.getId();
        this.year_desc = year.getDescr();
        this.year_year = year.getYear();
        this.nation = nation;
        this.img_path = img_path;
        this.category = category;
        this.id = producer.getName() + "_" + year.getYear() + "_" + code;
    }

    public Set(String name, String code, Year year, Producer producer, String nation, String img_path, String category, boolean effectiveCode) {
        this.name = name;
        this.code = code;
        this.producer_id = producer.getId();
        this.producer_name = producer.getName();
        this.producer_color = producer.getColor();
        this.year_id = year.getId();
        this.year_desc = year.getDescr();
        this.year_year = year.getYear();
        this.nation = nation;
        this.img_path = img_path;
        this.category = category;
        this.id = producer.getName() + "_" + year.getYear() + "_" + code;
        this.effective_code = effectiveCode;
    }


    public Set() {
    }

    public String getName() {
        return name;
    }

    public String getCode() {
        return code;
    }

    public String getNation() {
        return nation;
    }

    public String getImg_path() {
        return img_path;
    }

    public String getId() {
        return id;
    }

    public String getProducer_name() {
        return producer_name;
    }

    public String getYear_id() {
        return year_id;
    }

    public String getProducer_color() {
        return producer_color;
    }

    public String getCategory() {
        return category;
    }

    public String getYear_desc() {
        return year_desc;
    }

    public int getYear_year() {
        return year_year;
    }

    public String getProducer_id() {
        return producer_id;
    }

    public boolean isEffectiveCode() {
        return effective_code;
    }
}

File: app/src/main/java/com/lucagiorgetti/surprix/model/Set.kt
package com.lucagiorgetti.surprix.model

/**
 * Created by Utente on 17/04/2017.
 */
class Set {
    var id: String? = null
        private set
    var code: String? = null
        private set
    var name: String? = null
        private set
    var year_id: String? = null
        private set
    var year_desc: String? = null
        private set
    var year_year = -1
        private set
    var producer_id: String? = null
        private set
    var producer_name: String? = null
        private set
    var producer_color: String? = null
        private set
    var thanks_to: String? = null
        private set
    var nation: String? = null
        private set
    var img_path: String? = null
        private set
    var category: String? = null
        private set
    var isEffectiveCode = true
        private set


    constructor(name: String?, code: String, year: Year, producer: Producer, nation: String?, img_path: String?, category: String?) {
        this.name = name
        this.code = code
        producer_id = producer.id
        producer_name = producer.name
        producer_color = producer.color
        year_id = year.id
        year_desc = year.descr
        year_year = year.year
        this.nation = nation
        this.img_path = img_path
        this.category = category
        id = producer.name + "_" + year.year + "_" + code
    }

    constructor(name: String?, code: String, year: Year, producer: Producer, nation: String?, img_path: String?, category: String?, effectiveCode: Boolean) {
        this.name = name
        this.code = code
        producer_id = producer.id
        producer_name = producer.name
        producer_color = producer.color
        year_id = year.id
        year_desc = year.descr
        year_year = year.year
        this.nation = nation
        this.img_path = img_path
        this.category = category
        id = producer.name + "_" + year.year + "_" + code
        isEffectiveCode = effectiveCode
    }

    constructor()
}